### ISSUE TITLE
Easy way to separate all colors?

### ISSUE THREAD

**Issue by DinsmoreDesign** [jgthms/bulma#1284]
This is about **Bulma**.

### Overview of the problem

This is about the Bulma **CSS framework**
I'm using Bulma **version** [0.6.0]
My **browser** is: Chrome/Firefox/Edge/IE 9+

### Description

We're building out a proof of concept to go into all of our internal applications that would load the base version of our Bulma build by default on all servers and then using a JavaScript method to do an AJAX call to our configuration service and pull back client-specific items, such as CSS (colors), names and logos based on the URL they're hitting the sever at. Everything has been tested and seem to work nicely enough, the only problem is replacing colors inside Bulma.

I'm using SASS to configure our CSS build, but it doesn't look like there's an easy way to build a file that only replaces colors, rather than build the entire CSS library out. It seems the only way to do this is to manually go into each file, find any mention of a color and separate it out into my own folder structure... Obviously this is tedious, but if it's the way it has to be done, so be it... I was just curious if there was an easier way to do this, so I can save myself some headache?

tldr; I want to build all of Bulma as a base (including colors), insert it as a static link in the head of our applications and then use a second, dynamically called, CSS file to replace all the base colors, without having to overwrite the entire first library. Is this easily doable?

**Comment by jgthms** [comment]
It seems that what you need is CSS Variables: https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables

Then you can customize your Bulma setup with your own variables like `$blue: var(--blue)`. Then, to override that blue color, you only need to set it at the root:

```
:root {
  --blue: #00f;
}
```

**Comment by DinsmoreDesign** [comment]
@jgthms - I didn't even think of CSS Variables! That would be an awesome use case... the only problem is, we're anticipating we may have to support back to IE9 with some of our clients (I realize Bulma doesn't do this, I'm utilizing a simple percentage-based grid when targeting older browser) and CSS Variables aren't supported at all in IE. If we weren't dealing with paying customers and were only using this to brand our own things apps, I could care less about IE, since most people in our organization aren't using it, but it's harder to control a client's environment when we're not responsible for it :(

**Comment by DinsmoreDesign** [comment]
Just went ahead and separated all the colors in each file. Thankfully, it took a lot less time than I thought it would, since Bulma was built so modular! Looks to be working perfectly ;)
